src/category_theory/groupoid/subgroupoid.lean

```diff
@@ -537,6 +538,90 @@ lemma is_normal_map (hφ : function.injective φ.obj) (hφ' : im φ hφ = ⊤) (
 
 end hom
 
+section graph_like
+
+/-- A subgroupoid `is_graph_like` if it has at most one arrow between any two vertices. -/
+abbreviation is_graph_like := is_graph_like S.objs
+
+lemma is_graph_like_iff : S.is_graph_like ↔ ∀ (c d : S.objs), subsingleton (S.arrows c d) :=
+⟨ λ h c d, h c d, λ h c d, h c d ⟩
+
+end graph_like
+
+section disconnected
+
+/-- A subgroupoid `is_disconnected` if it has only isotropy arrows. -/
+abbreviation is_disconnected := is_disconnected S.objs
+
+lemma is_disconnected_iff : S.is_disconnected ↔ ∀ c d, (S.arrows c d).nonempty → c = d :=
+begin
+  split,
+  { rintro h c d ⟨f,fS⟩,
+    rw ←@subtype.mk_eq_mk _ _ c (mem_objs_of_src S fS) d (mem_objs_of_tgt S fS),
+    exact h ⟨c,mem_objs_of_src S fS⟩ ⟨d,mem_objs_of_tgt S fS⟩ ⟨⟨f,fS⟩⟩, },
+  { rintros h ⟨c,hc⟩ ⟨d,hd⟩ ⟨f,fS⟩,
+    simp only [subtype.mk_eq_mk],
+    exact h c d ⟨f,fS⟩, },
+end
+
+/-- The isotropy arrows of `S` -/
+inductive disconnect.arrows : Π (c d : C), (c ⟶ d) → Prop
+| mk {c : C} {γ : c ⟶ c} (hγ : γ ∈ S.arrows c c) : disconnect.arrows c c γ
+
+/-- The isotropy subgroupoid of `S` -/
+def disconnect : subgroupoid C :=
+{ arrows := disconnect.arrows S,
+  inv := by { rintros _ _ _ ⟨h⟩, constructor, apply S.inv h, },
+  mul := by { rintros _ _ _ _ ⟨h⟩ _ ⟨h'⟩, constructor, apply S.mul h h', } }
+
+lemma disconnect_le : (S.disconnect) ≤ S :=
+by {rw le_iff, rintros _ _ _ ⟨⟩, assumption, }
+
+lemma disconnect_normal (Sn : S.is_normal) : S.disconnect.is_normal :=
+{ wide := λ c, by { constructor, exact Sn.wide c, },
+  conj := λ c d p γ hγ, by { constructor, apply Sn.conj, cases hγ, assumption, } }
+
+lemma mem_disconnect_iff {c d : C} (f : c ⟶ d) :
+  f ∈ S.disconnect.arrows c d ↔ (c = d ∧ f ∈ S.arrows c d) :=
+begin
+  split,
+  { rintro ⟨⟩, split, refl, assumption, },
+  { rintro ⟨rfl,_⟩, constructor, assumption, },
+end
+
+end disconnected
+
+section full
+
+variable (D : set C)
+
+/-- The arrows of the full groupoid on a set `D : set C` -/
+inductive full.arrows : Π (c d : C), (c ⟶ d) → Prop
+| mk {c d : C} (hc : c ∈ D) (hd : d ∈ D) (γ : c ⟶ d) : full.arrows c d γ
```

Comment:
Are you sure you want this to be an inductive? The following definition should be equivalent and easier to use.
```lean
def disconnect.arrows (c d : C) (f : c ⟶ d) : Prop :=
c ∈ D ∧ d ∈ D
```

Reply:
Probably needs a `@[nolint unused_arguments]` :) With this simple definition you no longer need `mem_full_(arrows_)iff` right?

Reply:
If this approach is used, I would prefer it to be inlined in the definition of `full`, and simps would generate the desired lemma.

Reply:
Yeah, actually, the inductive makes things more painful in this case. I guess I just have an "everything inductive" phase which I should overcome :)